web_external/Datasets/ApplyMetadataView.js

```diff
@@ -0,0 +1,261 @@
+// Model for a metadata file
+isic.models.MetadataFileModel = girder.Model.extend({
+    name: function () {
+        var time = girder.formatDate(this.get('time'), girder.DATE_SECOND);
+        var fileName = this.get('file').name();
+        var userName = this.get('user').name();
+        return time + ' \u2014 ' + fileName + ' \u2014 ' + userName;
+    }
+});
+
+// Collection of metadata files as returned by of DatasetModel.getRegisteredMetadata()
+isic.collections.MetadataFileCollection = girder.Collection.extend({
+    model: isic.models.MetadataFileModel,
+
+    // Sort in reverse chronological order
+    sortField: 'time',
+    sortDir: girder.SORT_DESC,
+
+    parse: function (data) {
+        _.each(data, function (item) {
+            item.file = new girder.models.FileModel(item.file);
+            item.user = new isic.models.UserModel(item.user);
+
+            // Use file ID as item ID
+            item._id = item.file.id;
+        });
+        return data;
+    }
+});
+
+// Model for a single metadata error
+isic.models.MetadataErrorModel = girder.Model.extend({
```

Comment:
In this case, it's probably better as a `Backbone` model. I will update the code.